docs/celery.rst

```diff
@@ -111,7 +118,48 @@ For error handling update the task state to failure and provide errors:
         except SomeError as err:
             errors = [err]
             update_task_state(my_cool_task, states.FAILURE, {'errors': errors})
-            raise
+            raise Ignore()
+
+Option 2
+--------
+
+This option raises an exception which celery does not catch.
+Soil will catch this and set the error to the error message in the exception.
+The resulting task will be marked as a failure meaning ``task.failed()`` will return ``True``
+If calling with ``CELERY_ALWAYS_EAGER = True`` (i.e. a dev environment), the exception will "bubble up" to the calling code.
+
+.. code-block:: python
+
+    from soil import DownloadBase
+    from soil.progress import update_task_state
+    from soil.util import expose_cached_download
+
+    @task
+    def my_cool_task():
+        try:
+            # do some stuff
+        except SomeError as err:
+            errors = [err]
+            raise SomeError(errors)
```

Comment:
Why catch and re-raise with a list? I think the `try`/`except` should either be removed or this example should do something meaningful in the `except` clause. Would this be reasonable:
```py
@task
def my_cool_task():
    # do some stuff
    raise SomeError("my uncool error")
```